runtime/gc_base/StringTable.cpp

```diff
@@ -623,7 +630,12 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 
 		if (isASCII) {
 			for (UDATA i = 0; i < length; ++i) {
-				hash = (hash << 5) - hash + data[i];
+				U_8 c = data[i];
+				if (0xC0 == c) {
+					c = 0;
+					i++;
```

Comment:
Please use `i += 1;` like line 603; same for lines 730, 745, 773 and 784.

Reply:
Updating again to tweak the comment wording.